pyboy/core/cartridge/cartridge.py

```diff
@@ -57,7 +57,15 @@ def load_romfile(filename):
         romdata = array("B", romfile.read())
 
     logger.debug(f"Loading ROM file: {len(romdata)} bytes")
+    if len(romdata) == 0:
+        logger.error("ROM file is empty!")
+        raise Exception("Empty ROM file")
+
     banksize = 16 * 1024
+    if len(romdata) % banksize != 0:
+        logger.error("Unexpected ROM file length")
+        raise Exception("Bad ROM file size")
```

Comment:
Exceptions are not working so well in Cython, but this isn't a central part of any important logic, so let's just leave it like this.

Reply:
They seemed okay when I tested them, but if it becomes a problem some day we can deal with it. It's not likely that these will fire for anyone--but I made a mistake once that caused a rom to have 0 length, and this is easier to understand than a segfault, haha.